torch_frame/data/multi_nested_tensor.py

```diff
@@ -210,6 +220,95 @@ def _row_index_select(self, index: Tensor) -> Tensor:
+        if dim == 0 or dim == -3:
+            index = self._to_positive_index(index, dim=0)
+            start_idx = index * self.num_cols
+            end_idx = (index + 1) * self.num_cols + 1
+            offset = self.offset[start_idx:end_idx]
+            values = self.values[offset[0]:offset[-1]]
+            offset = offset - offset[0]
+            return MultiNestedTensor(num_rows=1, num_cols=self.num_cols,
+                                     values=values, offset=offset)
+        elif dim == 1 or dim == -2:
+            index = self._to_positive_index(index, dim=1)
+            start_idx = torch.arange(index, self.num_rows * self.num_cols,
+                                     self.num_cols, device=self.device)
+            diff = self.offset[start_idx + 1] - self.offset[start_idx]
+            batch, arange = batched_arange(diff)
+            # Compute values
+            values = self.values[self.offset[start_idx][batch] + arange]
+            # Compute offset
+            offset = diff.new_zeros(diff.numel() + 1)
+            torch.cumsum(diff, dim=0, out=offset[1:])
+            return MultiNestedTensor(num_rows=self.num_rows, num_cols=1,
+                                     values=values, offset=offset)
+        else:
+            raise RuntimeError(f"Unsupported dim={dim} for index_select.")
+
+    def _to_positive_index(
+        self,
+        index: Union[int, Tensor],
+        dim: int,
+        is_slice_end: bool = False,
+    ):
+        """Helper function to map negative indices to positive indices and
+        raise :obj:`IndexError` when necessary.
+
+        Args:
+            index: Union[int, Tensor]: Input :obj:`index` with potentially
+                negative elements.
+            is_slice_end (bool): Whether a given index (int) is slice or not.
+                If :obj:`True`, we have more lenient :obj:`IndexError`.
+                (default: :obj:`False`)
+        """
+        assert dim in [0, 1]
+        if isinstance(index, int):
+            if dim == 0:
+                if index < 0:
+                    index = index + self.num_rows
+                if is_slice_end:
+                    if index < 0 or index > self.num_rows:
+                        raise IndexError("Row index out of bounds!")
+                else:
+                    if index < 0 or index >= self.num_rows:
+                        raise IndexError("Row index out of bounds!")
+            else:
+                if index < 0:
+                    index = index + self.num_cols
+                if is_slice_end:
+                    if index < 0 or index > self.num_cols:
+                        raise IndexError("Column index out of bounds!")
+                else:
+                    if index < 0 or index >= self.num_cols:
+                        raise IndexError("Column index out of bounds!")
+        elif isinstance(index, Tensor):
+            assert not is_slice_end
+            index = index.clone()
```

Comment:
nit: In most cases, the clone here is unnecessary:
```python
neg_index = index[neg_idx]
if neg_index.numel() > 0:
    index = index.clone()
    index[neg_idx] = max_entries + neg_index

Reply:
That's a good point. Fixed.